tests/test_evaluation/test_split_evaluation.py

```diff
@@ -305,3 +295,70 @@ def test_multilingual_one_missing_lang_in_one_split(
     # output merged result with previous results
     assert results[0].scores.keys() == {"test", "val"}
     assert len(results[0].scores["test"]) == 2
+
+
+def test_all_splits_evaluated_with_overwrite(model, tasks, tmp_path):
+    evaluation = MTEB(tasks=tasks)
+    results = evaluation.run(
+        model,
+        eval_splits=["val"],
+        output_folder=str(tmp_path / "all_splits_evaluated_with_overwrite"),
+        verbosity=2,
+    )
+
+    assert "MockRetrievalTask" == results[0].task_name
+    last_evaluated_splits = evaluation.get_last_evaluated_splits()
+    assert set(last_evaluated_splits["MockRetrievalTask"]) == {"val"}
+    assert len(last_evaluated_splits["MockRetrievalTask"]) == 1
+    assert results[0].scores.keys() == {"val"}
+
+    results2 = evaluation.run(
+        model,
+        eval_splits=["val"],
+        output_folder=str(tmp_path / "all_splits_evaluated_with_overwrite"),
+        verbosity=2,
+        overwrite_results=True,
+    )
+    assert "MockRetrievalTask" == results2[0].task_name
+    last_evaluated_splits = evaluation.get_last_evaluated_splits()
+    assert set(last_evaluated_splits["MockRetrievalTask"]) == {"val"}
+    assert len(last_evaluated_splits["MockRetrievalTask"]) == 1
```

Comment:
Let's swap the order here - check len first before asserting values. Same above.